dist/ExtUtils-ParseXS/lib/perlxs.pod

```diff
@@ -1,2065 +1,4706 @@
+                       // over one or more lines:
+                     | "ATTRS:" [foo bar baz]
+
+
+ alias             = "ALIAS:"
+                       // One or more lines; each with zero or more
+                       // {alias_name, op, index} triplets:
+                       (
+                         [bar]       "="  [5]
+                       | [Foo::baz]  "="  [A_CPP_DEFINE]
+                       | [Bar::boz]  "=>" [Foo::baz]
+                       )*
+
+=head1 OVERVIEW OF XS AND XSUBS
+
+=head2 Initial and Further Reading
+
+This document is structured on the assumption that you are already
+familiar with the very basics of XS and XSUBs; in particular, the code
+examples may make use of common keywords that are only described later in
+the file. But once you have that basic familiarity, then this document
+may be read through in order.
+
+It is in two main parts. First there is a long overview part, which
+explains (in great detail) what XS and XSUBs are, how the perl interpreter
+calls XSUBs, and how data is passed to and from an XSUB. There is much
+more detail here than is strictly necessary for writing simple XSUBs, but
+this document is intended to be comprehensive. Then comes the reference
+manual proper, which has a section for each keyword and other parts of an
+XSUB declaration and definition, plus a few more general topics, such as
+using typemaps and storing static data.
+
+If necessary, read L<perlxstut> first for a gentler tutorial introduction.
+In addition, you may find the following Perl documents useful.
+
+=over
+
+=item *
+
+L<perlxstypemap>: this describes typemap files and how to create new
+typemaps. These are code templates which are used by the XS compiler to
+automatically generate code which converts between Perl and C data types.
+Creating an interface to a C library is sometimes mainly a case of adding
+new typemap entries to handle the new data types which the library uses.
+
+=item *
+
+L<perlguts>: this contains details of selected parts of the internals of
+the Perl interpreter. A better understanding of that may help when writing
+more complex XSUBs, or when debugging.
+
+Note that much of the L<data passing overview|/Overview of how data is
+passed to and from an XSUB> part of this document is just a summary of
+the parts of of perlguts which are most relevant to writing XS code,
+possibly saving you from having to actually read that document.
+
+=item *
+
+L<perlclib>: this describes how and when to use the basic C and OS library
+functions from XS. Often, the Perl API contains functions which you should
+use I<instead> of the standard C library ones, e.g. using C<PerlIO_read()>
+instead of C<fread()>.
+
+=item *
+
+L<perlcall>: this describes how to call Perl functions and do the
+equivalent of C<eval ""> from C.
+
+=item *
+
+L<perlembed>: this describes how to embed a complete Perl interpreter
+within another application.
```

Comment:
perlcall barely mentions eval_pv, perlembed provides much better documentation for eval_pv/eval_sv.